apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -444,6 +493,102 @@ public Mono<Void> insert(@NonNull Span span) {
                 .then();
     }
 
+    @Trace(dispatcher = true)
+    public Mono<Void> batchInsert(@NonNull List<Span> spans) {
+        return Mono.from(connectionFactory.create())
+                .flatMapMany(connection -> {
+                    if (spans.isEmpty()) {
+                        return Mono.just(0L);
+                    }
+
+                    return insert(spans, connection);
+                })
+                .then();
```

Comment:
Let's better return the aggregated sum of inserted records, even if we don't use it at the beginning. This is also in contradiction with the case for spans being empty (although I'm recommending to remove that).